Code/Tools/AssetProcessor/native/ui/BuilderDataItem.cpp

```diff
@@ -60,17 +65,33 @@ namespace AssetProcessor
         return m_parent;
     }
 
+    AZStd::shared_ptr<BuilderDataItem> BuilderDataItem::InsertChild(AZStd::shared_ptr<BuilderDataItem>&& itemToBeInserted)
+    {
+        //! It's the caller's responsibility to ensure the builder has a unique name
+        if (m_itemType != ItemType::InvisibleRoot ||
+            (itemToBeInserted->GetItemType() != ItemType::InvisibleRoot && itemToBeInserted->GetItemType() != ItemType::Builder) ||
+            itemToBeInserted->GetParent().lock().get() != this || m_childNameToIndex.contains(itemToBeInserted->GetName()))
+        {
+            return nullptr;
+        }
+
+        m_children.push_back(itemToBeInserted);
+        m_childNameToIndex[itemToBeInserted->GetName()] = aznumeric_cast<int>(m_children.size() - 1);
```

Comment:
`itemToBeInserted` should be moved to avoid a copy.
```suggestion
        m_childNameToIndex[itemToBeInserted->GetName()] = aznumeric_cast<int>(m_children.size() - 1);
        m_children.emplace_back(AZStd::move(itemToBeInserted));
```

Reply:
I will do that. Thanks for the suggestion.

Reply:
updated